src/Microsoft.Graph.Core/Tasks/LargeFileUploadTask.cs

```diff
@@ -72,14 +74,10 @@ private IUploadSession ExtractSessionFromParsable(IParsable uploadSession)
             if (!uploadSession.GetFieldDeserializers().ContainsKey("uploadUrl"))
                 throw new ArgumentException("The Parsable does not contain the 'uploadUrl' property");
 
-            var uploadSessionType = uploadSession.GetType();
-
-            return new UploadSession()
-            {
-                ExpirationDateTime = uploadSessionType.GetProperty("ExpirationDateTime").GetValue(uploadSession, null) as DateTimeOffset?,
-                NextExpectedRanges = uploadSessionType.GetProperty("NextExpectedRanges").GetValue(uploadSession, null) as List<string>,
-                UploadUrl = uploadSessionType.GetProperty("UploadUrl").GetValue(uploadSession, null) as string
-            };
+            // convert to local type as we don't have the type info for the upload session just that it implements IParsable
+            var uploadSessionStream = KiotaSerializer.SerializeAsStream(CoreConstants.MimeTypeNames.Application.Json, uploadSession, false);// just in case there's a backing store 
+            var uploadSessionJsonNode = new JsonParseNode(JsonDocument.Parse(uploadSessionStream).RootElement);
```

Comment:
If there is a parse async method, I'm runtime will not yell at people when in aspnet core contexts

Reply:
Just to confirm, do you mean we use `JsonDocument.ParseAsync`?

Reply:
yes

Reply:
As this method is called/used by the constructor, using the `ParseAsync` may not be ideal here. (It will force us to do something like `GetAwaiter.GetResult` as constructors can't be async).
Looking at the implementation of the Parse method, the stream is read into an ReadOnlyMemory/ArraySegment without having to do any task management which seems better creating strings or calling `GetAwaiter.GetResult`.

https://github.com/dotnet/runtime/blob/5535e31a712343a63f5d7d796cd874e563e5ac14/src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs#L120C13-L138C10

What do you think?

Reply:
We did move to using partial classes recently. I will take a look at this and update...

Reply:
@baywet The latest updates here will mark the constructor with the IParsable obsolete going forward. 

What this means is that we'll add a new constructor that takes in the `IUploadSession` interface so there won't be any need for serializing/deserializing the members once these two updated to the service library are pulled in.

- https://github.com/microsoftgraph/msgraph-beta-sdk-dotnet/pull/896
- https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/2653

The expectation really is that this path will only be realised in a scenario where a SDK user is using the latest version of the core package without an older version of the SDK models(which won't happen if they are using the core package via the service lib).

The diff is a bit noisy at the moment, so I'll mark this as ready for review once the cleanup work in https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/pull/909 is merged.

Reply:
Thanks for looking into this, it's a much better approach!
Since the model will implement both IUploadSession and IParsable, and since both constructors are identical beyond the first argument, I'm a bit worried the compiler might choose the first one and people will need to add casting to get the "right" one. Is this something that you ran into during testing?

Reply:
Actually, that's sort of what motivated the opening of https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/pull/909.

Once that is closed the diff will get much clearer and the tests should show that the model not using the `IUploadSession ` will lead to the obsolete usage warnings....